### ISSUE TITLE
Possible Not-So-Cas' situation: going through Lava Dive Room without Gravity Suit mandatory to complete

### ISSUE THREAD

**Issue by fxbx** [Vivelin/SMZ3Randomizer#117]
Seed 1977650472

Gravity Suit is in Upper Norfair, West - Ice Beam - it would be possible with mockball, though the Tricks option MockBall was not checked while generating the ROM [if it was checked I wouldn't be entering this in]
Speed Booster is in Lower Norfair, West - Missile (Gold Torizo) - though you can see it by entering via Misery Mire, apparently without TAS-grade glitches it is inaccessible
Therefore you'd have to enter Lower Norfair via Lava Dive Room to get the Speed Booster, which without Gravity Suit or Speed Booster is _non-trivial_

Spoiler log attached
[SMZ3_Cas_20220410-231501_1977650472.txt](https://github.com/Vivelin/SMZ3Randomizer/files/8469951/SMZ3_Cas_20220410-231501_1977650472.txt)

**Comment by Vivelin** [comment]
Thank you for reporting this! I'll look into this when I've got some more time.

**Comment by Vivelin** [comment]
I think you're expected to go there via the Worst Room in the Game and past the Mickey Mouse room. That should give you access to Gold Torizo without going through the Lava Dive Room.

Does that help?

**Comment by fxbx** [comment]
Yes, that would do it. I was afraid I was forgetting something here.

Issue may be closed, thank you.

**Comment by Vivelin** [comment]
No worries! Happy to help.

In the future I'd like to add a command like _"Hey tracker, how do I get to X (from Y)?"_, although this would take considerable effort as the SMZ3 logic doesn't have any "connections" between places.

**Comment by fxbx** [comment]
Watching Betus's VOD from last week - I might've asked to close too early. Space Jump needed to drain the lava - I'll check it out when I have the chance.

**Comment by Vivelin** [comment]
The check before Gold Torizo does take Space Jump into account, which you would've had access to according to your spoiler log.

**Comment by fxbx** [comment]
I see now, for some reason I thought all 3 weren't possible without glitch. I think this one is thoroughly dead now, thank you.

**Comment by Vivelin** [comment]
No problem. Thanks for taking the time to report this! This was non-trivial to figure out so there's clearly plenty of room for improvement.
